Cargo.toml

```diff
@@ -120,6 +120,9 @@ jj-lib = { path = "lib", version = "0.18.0" }
 jj-lib-proc-macros = { path = "lib/proc-macros", version = "0.18.0" }
 testutils = { path = "lib/testutils" }
 
+[patch.crates-io]
```

Comment:
comment: Haven't looked at the full PR in detail yet, but FYI, crates with patches cannot be published to [crates.io](https://crates.io), so leaving this in would potentially block the next release (if your change doesn't get merged upstream and released).

I forget what the specific behavior is if you try to `cargo publish` in those cases. It might silently ignore the patch and try to build without it, or it might just fail.

- If the former, it would probably be possible to add an off-by-default feature that builds the Watchman-with-trigger functionality.
- If we wanted to keep the patch indefinitely, we could probably specify that the source code should be fetched from your fork of the upstream repo rather than use `path` (and users could do their own testing by adding `path` back).

Conceivably, we could merge this code with all of the trigger notification code behind a feature _without_ configuring the patch in `Cargo.toml` directly, and require users who want to build it to also configure the patch (which can be done with `.cargo/config.toml` or a command-line option if they don't want to temporarily overwrite `Cargo.toml` itself).

Reply:
yeah, my plan is to just sit on the PR until that has been released and remove this part.

Reply:
radio silence on that PR from FB, so I made a file in JJ that contains the relevant extensions to the client api.  Should be very drop in to remove it when that is accepted.